src/hooks/useCurrentFaultTree.tsx

```diff
@@ -22,13 +23,20 @@ interface Props extends ChildrenProps {
 
 export const CurrentFaultTreeProvider = ({ faultTreeIri, children }: Props) => {
   const [_faultTree, _setFaultTree] = useState<FaultTree>();
+  const [rootReqProb, setRootReqProb] = useState<number | undefined>();
   const [showSnackbar] = useSnackbar();
 
   const fetchFaultTree = async () => {
-    faultTreeService
-      .find(faultTreeIri)
-      .then((value) => _setFaultTree(value))
-      .catch((reason) => showSnackbar(reason, SnackbarType.ERROR));
+    try {
+      const result = await faultTreeService.find(faultTreeIri);
+      _setFaultTree(result);
+      if (result.manifestingEvent.supertypes) {     
+        const reqProb = result.manifestingEvent.supertypes.hasFailureRate.requirement.upperBound
```

Comment:
Since you have definition:
```
export interface FaultEvent extends AbstractModel {
  eventType: EventType;
  name: string;
  description?: string;
  gateType?: GateType;
  probability?: number;
  children?: FaultEvent[];
  failureMode?: FailureMode;
  sequenceProbability?: number;
  childrenSequence?: any;
  rectangle?: Rectangle;
  probabilityRequirement?: number;
  iri?: string;
  supertypes?: {
    hasFailureRate?: {
      iri?: string;
      requirement?: {
        iri?: string;
        types?: string;
        upperBound?: number;
      };
    };
  };
}
```

Shouldn't you access it as follows?:
...supertypes?.hasFailureRate? ...

Reply:
That's true, my bad. I was fighting with this context and didn't mention it. Also fixed